fairscale/optim/grad_scaler.py

```diff
@@ -3,20 +3,57 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
-import logging
-from typing import Any, Dict
+from collections import abc, defaultdict
+from enum import Enum
+from typing import Any, Dict, List, Optional, Union
+import warnings
 
 import torch
+from torch.cuda import FloatTensor  # type: ignore
 from torch.cuda.amp import GradScaler as TorchGradScaler
+from torch.cuda.amp.common import amp_definitely_not_available
 import torch.distributed as dist
 from torch.optim import Optimizer
+from torch.optim.sgd import SGD
 
-from .oss import OSS
+
+class _GeneralMultiDeviceReplicator(object):
```

Comment:
Would be great if we could link the PyTorch class for reference here.

Reply:
done